File: Java/P5/Civitas/src/civitas/CasillaCalle.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

import java.util.ArrayList;

/**
 *
 * @author arturo
 */
public class CasillaCalle extends Casilla{
    private tituloPropiedad titulo;
    private float importe;
    
    CasillaCalle(tituloPropiedad titulo){
        super(titulo.getNombre());
        this.titulo=titulo;
        importe=titulo.getPrecioCompra();
    }
    
    @Override
    public void recibeJugador(int actual, ArrayList<Jugador> todos){ 
        if(jugadorCorrecto(actual,todos)==true){
                informe(actual,todos);
                Jugador jugador=todos.get(actual);
            if(titulo.tienePropietario()==false){
                jugador.puedeComprarCasilla();
            }else{
                titulo.tramitarAlquiler(jugador);
            }
        }
    }
    
    tituloPropiedad getTituloPropiedad(){
        return titulo;
    }
    
    @Override
    public String toString(){
        return "Importe a pagar en esta casilla: "+this.importe+"\n";
    }
}

File: Ruby/Civitas/lib/casilla_calle.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

module Civitas
  class CasillaCalle < Casilla
    def initialize(titulo)
      super (titulo.nombre)
      @titulo_propiedad=titulo
    end

    # Método sobrescrito
    def recibe_jugador(actual, todos)
      if(jugador_correcto(actual, todos))
        informe(actual, todos)
        jugador=todos[actual]
        
        unless(@titulo_propiedad.tiene_propietario)
          jugador.puede_comprar_casilla
          @titulo_propiedad.tramitar_alquiler(jugador)
        end
        
      end
    end
    
    def to_s
      "\n   **DESCRIPCIÓN CASILLA**
       \n\tTipo: CALLE
       \n\tNombre: #{nombre}
       \n\tPropiedad: #{@titulo_propiedad.to_s}"
    end
    
  end
end